covid_model/model_fit.py

```diff
@@ -97,13 +97,13 @@ def run(self, engine, method='curve_fit', window_size=14, look_back=None,
             t1 = perf_counter()
             print(f'Transmission control fit {i + 1}/{len(trim_off_end_list)} completed in {t1 - t0} seconds.')
             if write_batch_output:
-                model.specifications.tags['run_type'] = 'intermediate-fit'
-                model.specifications.write_to_db(engine)
+                model.tags['run_type'] = 'intermediate-fit'
+                model.write_to_db(engine)
 
         self.fitted_tc = tc
         self.fitted_tc_cov = fitted_tc_cov
         self.fitted_model = model
-        self.fitted_specs = model.specifications
+        self.fitted_specs = model
```

Comment:
this messes with me. as long as `self.fitted_specs` is always treated as an instance of `CovidModelSpecifications` and never treated as an instance of `CovidModel` then this seems fine. But it relies on the programmer's obeying that constraint. Seems like a necessary consequence of duck typing. Or am I way off base?